scripts/base/protocols/ssh/main.zeek

```diff
@@ -54,6 +54,9 @@ export {
 		host_key_alg:    string       &log &optional;
 		## The server's key fingerprint
 		host_key:        string       &log &optional;
+		## Indicate whether there is a discrepancy between the host that
+		## set up the TCP connection, and sets up the SSH key exchange.
+		reversed_conn:   bool         &log &optional;
```

Comment:
Somewhat delayed, but the other question that comes up with this change is whether this field is something we want to add to ssh.log by default.

It seems for most deployments this field will always be blank. Once it's `T`, it's of course interesting and very valuable, but generally it's entropy is extremely low.

To me it sounds more appropriate to make this a notice (or weird with action notice). This may also match better with existing workflows monitoring notices/weirds, rather than having "features" in the logs.

Adding it to ssh.log can of course be done by additional scripts by extending the log, it's just that the default scripts wouldn't include this field.

Other thoughts?

Reply:
> To me it sounds more appropriate to make this a notice (or weird with action notice). This may also match better with existing workflows monitoring notices/weirds, rather than having "features" in the logs.

My thoughts too.

Reply:
Thanks for the feedback! I was also thinking whether the correct place for this would be in the `base` scripts, or the `policy` ones, didn't think of a notice/weird log yet as we use it in the `ssh.log` internally. The current commit does not sound like the best solution indeed, so I think that leaves us with a few possible solutions:

1. Change the `base/protocols/ssh/main.zeek` script and let it create a `notice.log` entry
2. Create a `policy/protocols/ssh/detect-reverse-ssh.zeek` script and let it create an optional feature in the `ssl.log`
3. Create a `policy/protocols/ssh/detect-reverse-ssh.zeek` script and let it create a feature in the `notice.log`

My personal suggestion is to do both 1 and 2; enable the detection in the notice log by default, but also optionally write the field to an `ssl.log`. What do you think?

Reply:
> Create a policy/protocols/ssh/detect-reverse-ssh.zeek script and let it create a feature in the notice.log
> My personal suggestion is to do both 1 and 2; enable the detection in the notice log by default, but also optionally write the field to an ssl.log. What do you think?

I'd actually steer towards just doing 3) and loading the new script by default via `site/local.zeek`. There aren't notices raised from `base/protocols/*` currently so that would be new about 1). It's unclear if 2) really needs to be in the base distribution.

There's certainly also the argument that the notice could be added through an external `zkg` "detection package". However, given how closely it's connected with the new event(s) and pretty self-contained, adding it directly to `policy` seems reasonable to me for this one.

Reply:
Thanks again! Going to have a look at it today

Reply:
Just to add a thought that's implicit in the above: we could do none of these here, and instead add the field in a Zeek package for the folks eager to have it. We're reluctant to add anything to the policy folder at this point (says the guy who dumped the whole management framework in there — forgive the irony!).

Reply:
I like @ckreibich 's suggestion.

Reply:
Yes, then I'll move the script to a separate package.

Reply:
Binpac/analyzer changes would stay in this repo. It'd also be great to keep the pcap and the test. The latter would needs some changes - maybe handling the new events directly and just printing.

Reply:
Cool. Closing this PR for now and going to submit a new PR with the feedback applied. Thanks for all the feedback!